kit/ChildSession.hpp

```diff
@@ -100,7 +100,7 @@ class ChildSession final : public Session
         return _docManager->sendFrame(msg.data(), msg.size(), WSOpCode::Binary);
     }
 
-    bool sendProgressFrame(const char* id, const std::string &jsonProps);
+    bool sendProgressFrame(const char* id, const std::string &jsonProps, std::string forcedID = "");
```

Comment:
you can make this: const std::string& forcedID = "" and "const std::string& forcedID" above to match.